### DISCUSSION TITLE
Setting max_iter for ipopt

### DISCUSSION THREAD

**Discussion by gioperin** [do-mpc/do-mpc#200] [Q&A] [answered]
Hello,

I tried to set the keyword `max_iter` to `nlpsol_opts` in the method `mpc.set_param()`, obtaining a keyword error. The keyword is one of the general keywords supported by casadi solvers, but it is not specific of ipopt. It seems that with the current do-mpc interface it is impossible to set general keywords, but only ipopt-specific ones. Is there a workaround for this?

Thanks,

Giovanni

**Comment by zmdntnu** [comment, marked as answer]
Hi,
I think you can call 'ipopt.max_iter':, and put it into the setting under 'nlpsol_opts' inside setup_mpc.
Example:
setup_mpc = {  
  'nlpsol_opts': {'ipopt.max_iter': 1000}
}

**Comment by 4flixt** [comment]
Thankyou @zmdntnu , that is exactly the way to go. This ``'nlpsol'`` keyword takes a dict of arguments that is passed to the chosen solver.
